### DISCUSSION TITLE
What's Quasar's QTree expecting after 'lazy-load'?

### DISCUSSION THREAD

**Discussion by j4nSolo** [zauberzeug/nicegui#3141] [Q&A]
### Question

I want to create a lazy-loading tree and have added a listener to its `lazy-load` event. Whenever I click on an expandable icon, the listener gets called normally but the loading animation never changes, as if it was expecting some confirmation. My final goal is to load children based on the parent that's expanded. I did a bit of debugging but I don't see the handler expecting any return nor I see any method I may call.

On Quasar's website, the example uses javascript to handle it and its `onLazyLoad` function does receive a couple callbacks (`done `and `fail`) that serve the behaviour I'm missing: https://quasar.dev/vue-components/tree#lazy-loading

Here's a minimal reproducible example:

```py
from nicegui import ui

def callback(tree, args):
    print(tree)
    print(args)

tree = ui.tree(
    [{'name': url.get('text'), 'url': url.get('href'), 'lazy': True}
     for url in (
         {'text': 'Google', 'href': 'https://google.com'},
         {'text': 'Youtube', 'href': 'https://youtube.com'}
     )],
    node_key='url',
    label_key='name',
)

tree.on('lazy-load', lambda e: callback(tree, e.args))

ui.run()
```

Moreover, how would I add the children once I figure this out?

**Comment by falkoschindler** [comment]
Hi @j4nSolo,

The event handler needs to call either `done` with a list of children or `fail`. You can do that with a plain JavaScript handlers:
```py
tree.on('lazy-load', js_handler='(details) => details.done([{name: "Facebook", url: "https://facebook.com"}])')
```

But to retrieve real data from the server, you would probably need to make some HTTP request to a FastAPI endpoint.

**Reply by j4nSolo** [reply]
Hmmm, that makes it a bit too complicated. I went for another way without QTree so I guess I'll leave it here.
